coralproject/talk
# [CORL-3020]: Update to use pnpm instead of npm

## What does this PR do?

These changes update to use `pnpm` instead of `npm` as Coral's package manager.

To achieve this, it was necessary to:
- Update scripts to use `pnpm`
- Install some dependencies that weren't directly installed before
- Update `transformIgnorePatterns` in jest config to include `.pnpm` symlink path (See https://jestjs.io/docs/configuration#transformignorepatterns-arraystring and its `Tip` section)

In this PR, we also override `protobufjs` in our server `package.json` to address a security update needed there.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

First, you'll need to get set up with `pnpm`. You can run `npm install -g pnpm@8.14.3`
Then you'll want to clear your node modules. Run `sh scripts/clear-node_modules.sh`
Then you can run `sh scripts/pnpm-ci.sh` to install from the new `pnpm-lock.yaml`. (This is equivalent to `package-lock.json`)
Then you can do all the usual things. 
`pnpm run watch` inside client and server to run Coral
(Might need to run `sh scripts/build.sh` first if you see linting errors)
`pnpm run lint` inside client and server to check linting (or `sh scripts/lint.sh` to run linting everywhere)
`pnpm run test` inside client and server to check tests

## Were any tests migrated to React Testing Library?



## How do we deploy this PR?
